Posts/2021/10/tfl-secrets.md

```diff
@@ -0,0 +1,228 @@
+                                                         [Unregister-SecretVault, Unregister-SecretVault], [Get-SecretVault,   
+                                                         Get-SecretVault], [Set-SecretVaultDefault, Set-SecretVaultDefault],   
+                                                         [Test-SecretVault, Test-SecretVault], [Set-Secret, Set-Secret],       
+                                                         [Set-SecretInfo, Set-SecretInfo], [Get-Secret, Get-Secret],
+                                                         [Get-SecretInfo, Get-SecretInfo], [Remove-Secret, Remove-Secret],
+                                                         [Unlock-SecretVault, Unlock-SecretVault]}
+    Binary 1.0.5   Microsoft.PowerShell.SecretStore      {[Unlock-SecretStore, Unlock-SecretStore], [Set-SecretStorePassword,  
+                                                         Set-SecretStorePassword], [Get-SecretStoreConfiguration,
+                                                         Get-SecretStoreConfiguration], [Set-SecretStoreConfiguration,
+                                                         Set-SecretStoreConfiguration], [Reset-SecretStore,
+                                                         Reset-SecretStore]}
+
+```
+
+As you can see, both modules have a number of commands you may need to use to manage secrets for your environment.
+
+## Registering and viewing a secret vault
+
+After you have the two modules installed, your next step is to register a secret vault.
+There are several vault options you can take advantage of, for this post, I'll use the built-in default vault.
+You configure the default vault like this:
+
+```powershell-console
+PS C:\Foo> # 4. Register the default secrets provider
+PS C:\Foo> $Mod = 'Microsoft.PowerShell.SecretStore'
+PS C:\Foo> Register-SecretVault -Name RKSecrets -ModuleName $Mod -DefaultVault
+PS C:\Foo> Get-SecretVault
+
+Name      ModuleName                       IsDefaultVault
+----      ----------                       --------------
+RKSecrets Microsoft.PowerShell.SecretStore True
+```
+Like the previous step, registering the vault does not create any output by default.
+You can view the vault you just created by using the `Get-SecretVault` command.
+
+## Setting a secret
+
+To create a new secret in your secret vault, you use the 1Set-Secret1 command, like this:
+
+```powershell-console
+
+PS> # 4. Register the default secrets provider
+PS> $Mod = 'Microsoft.PowerShell.SecretStore'
+PS> Register-SecretVault -Name RKSecrets -ModuleName $Mod -DefaultVault
+PS> # 5. View Secret vault
+PS> Get-SecretVault
+
+Name      ModuleName                       IsDefaultVault
+----      ----------                       --------------
+RKSecrets Microsoft.PowerShell.SecretStore True
+
+PS C:\Foo> # 6. Set the Admin password secret for Reskit forest
+PS C:\Foo> Set-Secret -Name ReskitAdmin -Secret 'Pa$$w0rd'
```

Comment:
You need to escape the backslash or get rid of the path in in the prompt. Since the path is not relevant, get rid of it.
```suggestion
PS> # 6. Set the Admin password secret for Reskit forest
PS> Set-Secret -Name ReskitAdmin -Secret 'Pa$$w0rd'
```